### PR TITLE
Add descriptions to all the tasks

### PR DESCRIPTION
**Pull request by RDIL**
Adds descriptions so people who execute `gradle tasks` will get descriptions of what all Loom's tasks will do.

### CODE REVIEW

**Review by natanfudge — APPROVED** [review]
Sure

File: src/main/java/net/fabricmc/loom/LoomGradlePlugin.java [hunk 1]
```diff
@@ -72,32 +72,38 @@ public void apply(Project target) {
 
 		TaskContainer tasks = target.getTasks();
 
-		tasks.register("cleanLoomBinaries", CleanLoomBinaries.class);
-		tasks.register("cleanLoomMappings", CleanLoomMappings.class);
+		tasks.register("cleanLoomBinaries", CleanLoomBinaries.class, t -> t.setDescription("Remove binary Jars created by loom."));
+		tasks.register("cleanLoomMappings", CleanLoomMappings.class, t -> t.setDescription("Remove mappings downloaded by loom."));
```
Comment (Juuxel) [inline src/main/java/net/fabricmc/loom/LoomGradlePlugin.java]:
```suggestion
		tasks.register("cleanLoomBinaries", CleanLoomBinaries.class, t -> t.setDescription("Removes binary jars created by Loom."));
		tasks.register("cleanLoomMappings", CleanLoomMappings.class, t -> t.setDescription("Removes mappings downloaded by Loom."));
```

File: src/main/java/net/fabricmc/loom/LoomGradlePlugin.java [hunk 2]
```diff
@@ -72,32 +72,38 @@ public void apply(Project target) {
...
 
 		tasks.register("cleanLoom").configure(task -> {
 			task.setGroup("fabric");
+			task.setDescription("Runs all loom cleanup tasks.");
 			task.dependsOn(tasks.getByName("cleanLoomBinaries"));
 			task.dependsOn(tasks.getByName("cleanLoomMappings"));
 		});
 
 		tasks.register("migrateMappings", MigrateMappingsTask.class, t -> {
+			t.setDescription("Migrate mappings to a new version.");
```
Comment (Juuxel) [inline src/main/java/net/fabricmc/loom/LoomGradlePlugin.java]:
```suggestion
			t.setDescription("Migrates mappings to a new version.");
```

File: src/main/java/net/fabricmc/loom/LoomGradlePlugin.java [hunk 3]
```diff
@@ -72,32 +72,38 @@ public void apply(Project target) {
...
 		TaskContainer tasks = target.getTasks();
 
-		tasks.register("cleanLoomBinaries", CleanLoomBinaries.class);
-		tasks.register("cleanLoomMappings", CleanLoomMappings.class);
+		tasks.register("cleanLoomBinaries", CleanLoomBinaries.class, t -> t.setDescription("Remove binary Jars created by loom."));
+		tasks.register("cleanLoomMappings", CleanLoomMappings.class, t -> t.setDescription("Remove mappings downloaded by loom."));
 
 		tasks.register("cleanLoom").configure(task -> {
 			task.setGroup("fabric");
+			task.setDescription("Runs all loom cleanup tasks.");
```
Comment (Juuxel) [inline src/main/java/net/fabricmc/loom/LoomGradlePlugin.java]:
```suggestion
			task.setDescription("Runs all Loom cleanup tasks.");
```

File: src/main/java/net/fabricmc/loom/LoomGradlePlugin.java [hunk 4]
```diff
@@ -72,32 +72,38 @@ public void apply(Project target) {
...
 			task.dependsOn(tasks.getByName("cleanLoomMappings"));
 		});
 
 		tasks.register("migrateMappings", MigrateMappingsTask.class, t -> {
+			t.setDescription("Migrate mappings to a new version.");
 			t.getOutputs().upToDateWhen((o) -> false);
 		});
 
 		tasks.register("remapJar", RemapJarTask.class, t -> {
+			t.setDescription("Remaps a Jar file.");
```
Comment (Juuxel) [inline src/main/java/net/fabricmc/loom/LoomGradlePlugin.java]:
```suggestion
			t.setDescription("Remaps the built project jar to intermediary mappings.");
```

File: src/main/java/net/fabricmc/loom/LoomGradlePlugin.java [hunk 5]
```diff
@@ -72,32 +72,38 @@ public void apply(Project target) {
...
 			t.getOutputs().upToDateWhen((o) -> false);
 		});
 
 		tasks.register("genSourcesRemapLineNumbers", RemapLineNumbersTask.class, t -> {
+			t.setDescription("Remaps line numbers for the genSources task.");
 			t.getOutputs().upToDateWhen((o) -> false);
 		});
 
 		tasks.register("genSources", t -> {
+			t.setDescription("Generate a human-readable version of the Minecraft source.");
```
Comment (Juuxel) [inline src/main/java/net/fabricmc/loom/LoomGradlePlugin.java]:
```suggestion
			t.setDescription("Generates a human-readable version of the Minecraft source.");
```

File: src/main/java/net/fabricmc/loom/LoomGradlePlugin.java [hunk 6]
```diff
@@ -143,35 +149,41 @@ public void apply(Project target) {
 			});
 		});
 
-		tasks.register("downloadAssets", DownloadAssetsTask.class);
+		tasks.register("downloadAssets", DownloadAssetsTask.class, t -> t.setDescription("Downloads required assets for Fabric."));
 
 		tasks.register("genIdeaWorkspace", GenIdeaProjectTask.class, t -> {
+			t.setDescription("Generate an IntelliJ IDEA workspace from this project.");
```
Comment (Juuxel) [inline src/main/java/net/fabricmc/loom/LoomGradlePlugin.java]:
```suggestion
			t.setDescription("Generates an IntelliJ IDEA workspace from this project.");
```

File: src/main/java/net/fabricmc/loom/LoomGradlePlugin.java [hunk 7]
```diff
@@ -143,35 +149,41 @@ public void apply(Project target) {
...
 			t.dependsOn("idea", "downloadAssets");
 			t.setGroup("ide");
 		});
 
 		tasks.register("genEclipseRuns", GenEclipseRunsTask.class, t -> {
+			t.setDescription("Generate Eclipse run configurations for this project.");
```
Comment (Juuxel) [inline src/main/java/net/fabricmc/loom/LoomGradlePlugin.java]:
```suggestion
			t.setDescription("Generates Eclipse run configurations for this project.");
```

File: src/main/java/net/fabricmc/loom/LoomGradlePlugin.java [hunk 8]
```diff
@@ -143,35 +149,41 @@ public void apply(Project target) {
...
 			t.dependsOn("downloadAssets");
 			t.setGroup("ide");
 		});
 
 		tasks.register("cleanEclipseRuns", CleanEclipseRunsTask.class, t -> {
+			t.setDescription("Remove Eclipse run configurations for this project.");
```
Comment (Juuxel) [inline src/main/java/net/fabricmc/loom/LoomGradlePlugin.java]:
```suggestion
			t.setDescription("Removes Eclipse run configurations for this project.");
```

File: src/main/java/net/fabricmc/loom/LoomGradlePlugin.java [hunk 9]
```diff
@@ -143,35 +149,41 @@ public void apply(Project target) {
...
 			t.setGroup("ide");
 		});
 
 		tasks.register("vscode", GenVsCodeProjectTask.class, t -> {
+			t.setDescription("Opens this project in VSCode.");
```
Comment (Juuxel) [inline src/main/java/net/fabricmc/loom/LoomGradlePlugin.java]:
```suggestion
			t.setDescription("Generates VSCode launch configurations.");
```

File: src/main/java/net/fabricmc/loom/LoomGradlePlugin.java [hunk 10]
```diff
@@ -143,35 +149,41 @@ public void apply(Project target) {
...
 			t.dependsOn("downloadAssets");
 			t.setGroup("ide");
 		});
 
-		tasks.register("remapSourcesJar", RemapSourcesJarTask.class);
+		tasks.register("remapSourcesJar", RemapSourcesJarTask.class, t -> t.setDescription("Remaps a source Jar."));
```
Comment (Juuxel) [inline src/main/java/net/fabricmc/loom/LoomGradlePlugin.java]:
```suggestion
		tasks.register("remapSourcesJar", RemapSourcesJarTask.class, t -> t.setDescription("Remaps the project sources jar to intermediary names."));
```

File: src/main/java/net/fabricmc/loom/LoomGradlePlugin.java [hunk 11]
```diff
@@ -143,35 +149,41 @@ public void apply(Project target) {
...
 
 		tasks.register("runClient", RunClientTask.class, t -> {
+			t.setDescription("Starts a development version of the Fabric client.");
```
Comment (Juuxel) [inline src/main/java/net/fabricmc/loom/LoomGradlePlugin.java]:
```suggestion
			t.setDescription("Starts a development version of the Minecraft client.");
```

File: src/main/java/net/fabricmc/loom/LoomGradlePlugin.java [hunk 12]
```diff
@@ -143,35 +149,41 @@ public void apply(Project target) {
...
 			t.dependsOn("jar", "downloadAssets");
 			t.setGroup("fabric");
 		});
 
 		tasks.register("runServer", RunServerTask.class, t -> {
+			t.setDescription("Starts a development version of the Fabric server.");
```
Comment (Juuxel) [inline src/main/java/net/fabricmc/loom/LoomGradlePlugin.java]:
```suggestion
			t.setDescription("Starts a development version of the Minecraft server.");
```

**Review by Juuxel — CHANGES_REQUESTED** [review]
Fixed some inconsistency with verbs (`generate` vs `generates` etc.) and some mistakes in the descriptions themselves.

**Review by Juuxel — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Comment by RDIL** [comment]
cc my friend @liach

**Pull request merged**